File: editor/en/[70]Climbing Stairs.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def climbStairs(self, n: int) -> int:
        previous = 1
        current = 1

        for _ in range(n - 1):
            previous, current = current, previous + current

        return current

# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/leetcode/editor/en/[70]Climbing Stairs.java
// leetcode submit region begin(Prohibit modification and deletion)
class Solution {
  public int climbStairs(int n) {
    if (n <= 2) {
      return n;
    }

    int[] dp = new int[n + 1];
    dp[1] = 1;
    dp[2] = 2;

    for (int i = 3; i <= n; i++) {
      dp[i] = dp[i - 1] + dp[i - 2];
    }

    return dp[n];
  }
}
// leetcode submit region end(Prohibit modification and deletion)
